Show current amount of items in queue as the bot's playing status

Arzte opened the pull request:
I'm tired of being asked how many items in the queue there are, so I made this, every ten minutes it refreshes in order to bring the best amounts.

Arzte committed:
Merge branch 'master' into develop

Arzte committed:
Semi-fix DM's not being sent

Arzte committed:
Use doublequotes so escaping is not needed.

Dougley committed:
this is an integer

Arzte committed:
Show number of reports in admin-queue as playing status

QOL for people who can see the admin-queue so that the plebs don't have to bother us with how many items are in the queue.

Arzte committed:
Merge remote-tracking branch 'origin/master' into develop

Arzte committed:
Lower time till it refreshes to a minute

I would go faster but there's no need to hard hit the db.
This is to ensure the amount that MB lists is the most correct, down to the minute.

SirDoctors commented:
Remember not to squash pr's originating from ``develop``

ThaTiemsz commented on index.js:
```diff
@@ -99,6 +101,12 @@ bot.Dispatcher.on(Events.GATEWAY_READY, () => {
+
+  // admin-queue amount playing status
+  setInterval(() => {
+    r.db('DFB').table('queue').distinct().count().run().then(c => bot.User.setGame(c.toString() + " reports in queue"))
```
Shouldn't this be filtered to only `adminMergeRequest` and `adminReviewDelete`? Because aren't [newCard](https://github.com/SteamingMutt/DiscordFeedback/blob/master/Commands/Commands/queue.js#L16) and [chatVote](https://github.com/SteamingMutt/DiscordFeedback/blob/master/Commands/Commands/queue.js#L40) also in the queue table?

If so then use something along the lines of: `.filter(r.row('type').eq('adminMergeRequest').or(r.row('type').eq('adminReviewDelete')))`

Correct me if I'm wrong, please @SteamingMutt

Arzte replied:
That's right, but that's the wrong solution to filtering it, I'd actually want to put in

.without('newCard').without('chatVote')

I'm away from my PC till morning (7:00-9:00AM EST) so I won't be able to fix it for the moment.

ThaTiemsz replied:
Ah, didn't know that was a thing, so:

`.without({ type: 'newCard' }).without({ type: 'chatVote' })`

Arzte replied:
That type bit is unneeded, you just need to pass the table name string, and that'll be enough.

Arzte committed:
Ignore feedback queue cards & votes from admin queue count

Totally didn't forget about these last night when I was really struggling to keep my eyes open.

Arzte committed:
Run game status on startup

otherwise you have to wait a whole ten minutes for it to show................

Arzte committed:
Bump time till a streak is invalid

This bumps it from 24 hours, to 36 hours, or 1 1/2 of a day. This should make peoples life a bit easier.

Arzte committed:
Revert "Bump time till a streak is invalid"

This reverts commit 11926f4c9aebe32680eeb7ba5c9c3339047a6830. I'll let someone else try to improve this... I can't seem to understand the code enough change it positivly for others without breaking everything.

Dougley reviewed (dismissed):
`.without()` is not the function you're looking for.

```
The opposite of pluck; takes an object or a sequence of objects, and returns them with the specified paths removed.
```

ThaTiemsz commented:
`.filter(r.row('type').eq('adminMergeRequest').or(r.row('type').eq('adminReviewDelete')))`
while ugly, this works 🙄

SirDoctors commented:
![pluck](https://user-images.githubusercontent.com/7450443/29998251-40779b80-8ff4-11e7-830c-025b9955b542.PNG)
![without](https://user-images.githubusercontent.com/7450443/29998252-508e0e28-8ff4-11e7-89e8-7e112eb549f3.PNG)
I think ``without()`` is exactly what I was looking for, I don't see why this wouldn't be what I want, it takes an object, (in this case the queue table) and returns it without ``newCard`` & ``chatVote``.

SteamingMutt commented:
`.without()` removes fields from the result, it doesn't filter them.

![image](https://user-images.githubusercontent.com/9768134/29998355-10021ed2-9029-11e7-9976-765380e47ec9.png)![](https://b2.dougley.com/C8ev29.png)

SirDoctors commented:
Ahh, so while it was actually the thing I was looking for, since my knowledge of the DB layout was wrong... will fix.

Arzte committed:
Merge remote-tracking branch 'origin/master' into develop

Arzte committed:
Properly get the amount of items in the admin queue

I might change the regex later on just to look for types with the word admin in it, but unless we decide to add more admin-queue types; this should be fine for now.

ThaTiemsz commented:
😄 
```js
r.db('DFB').table('queue').filter(doc => doc('type').match("^(adminMergeRequest|adminReviewDelete)$")).count().run().then(c => bot.User.setGame(c.toString() + " reports in queue"))
```

Arzte committed:
Remove unneeded function from copy-pasta code

*sigh* I'm beginning to feel like LW, I need to step off the high horse and double check everything to a fuller extent before pushing it.

SirDoctors commented:
Update this to current master and I'll merge this, thx.

nullpixel committed:
Merge branch 'master' into develop

SirDoctors commented:
Thx nill

Dougley commented on index.js:
```diff
@@ -1,5 +1,7 @@
 const Discordie = require('discordie')
 const UserVoice = require('uservoice-nodejs')
+const Dash = require('rethinkdbdash')
```
This needs to be centralized at some point, but it's beyond the scope of this PR

Dougley reviewed (changes requested):
I honestly see no point in a simple playing status, why is this needed again?

ThaTiemsz commented:
@SteamingMutt So that we don't need to ping bot devs anymore to ask how many items there are in the admin queue

nullpixel commented:
but the real question is, why do you even need to see how many items are in the admin queue?

LW001 commented:
Know when to spam dabbit to wake up and do queue things

nullpixel commented:
Dabbit literally said you can spam it whenever, it makes no difference

LW001 commented:
Well you can report stuff twice

### Does that make it high priority? Hell no

nullpixel commented:
that's a poor analogy

ThaTiemsz commented:
So we can roughly estimate when our merge requests/reports have been dealt with

nullpixel commented:
my concern if this is merged is that people will hold reports & dupes until a certain time, however

Thereatra commented:
I see this as a way to gameify things i.e "Oh there's only one more need before the queue hits a round number, might aswell find a dupe"

Arzte committed:
Fix regex when there's no name following the forum ID

Arzte committed:
Remove testing variable from testing regex

Arzte committed:
Bump time to remove reactions.
